src/utils/LegacyHelpers.ts

```diff
@@ -0,0 +1,111 @@
+import {
+  ImageStudioEvent,
+  ImageStudioEventType,
+  LegacySDKEvent,
+  SDKEvent,
+  SDKEventType,
+  SDKImageInputEventData,
+  SDKMetadata,
+} from '../types';
+
+/**
+ * translates LegacyImageStudioEvents to ImageStudioEvent
+ *
+ * @param data should be of type LegacyImageStudioEvent, but could also be any object
+ * @returns translated event, otherwise undefined signifies something we don't understand and therefore should not process
+ */
+export const translateLegacyImageStudioEvent = (
+  setUsingLegacyEventFormat: (param: boolean) => void,
```

Comment:
This field isn't mentioned in the doc block. Why is this a function rather than a boolean?

Reply:
just wraps it up in a setter to be explicit, rather than pushing a naked variable into this function and letting the code do whatever it pleases with it.